### PR TITLE
Define Github owner field pattern

### PR DESCRIPTION
**Pull request by mtojek**
## What does this PR do?

This PR defines the Github owner pattern for the manifest field.

## Why is it important?

We need to make sure that the field accepts private users or company teams.

## Checklist

- [x] I have added test packages to [`test/packages`](https://github.com/elastic/package-spec/tree/master/test/packages) that prove my change is effective.
- [x] I have added an entry in [`versions/N/changelog.yml`](https://github.com/elastic/package-spec/blob/master/versions/1/changelog.yml).

### CODE REVIEW

File: test/packages/bad_github_owner/changelog.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# newer versions go on top
+- version: "0.0.1"
+  changes:
+    - description: Initial draft of the package
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/0 # FIXME Replace with the real PR link
```
Comment (ycombinator) [inline test/packages/bad_github_owner/changelog.yml:6]: Replace this with https://github.com/elastic/package-spec/pull/164?

**Review by ycombinator — APPROVED** [review]
LGTM.

File: test/packages/bad_github_owner/changelog.yml (hunk 1 above)
Comment (mtojek) [reply test/packages/bad_github_owner/changelog.yml:6]: Not necessary, this package has been created with `elastic-package create package`. We can leave it here ;)

Comment (ycombinator) [reply test/packages/bad_github_owner/changelog.yml:6]: I know, just as a convention I've been using PRs where a test package was introduced as the link in that test package's changelog. You can see it in some other test packages. That way its easy to understand _why_ a test package was introduced (what was the context for it).

Comment (ycombinator) [reply test/packages/bad_github_owner/changelog.yml:6]: But this is not a hard requirement, just a convention. So if you want to leave this as-is, that's fine with me.

Comment (mtojek) [reply test/packages/bad_github_owner/changelog.yml:6]: Makes sense, let me update it quickly!

Comment (mtojek) [reply test/packages/bad_github_owner/changelog.yml]: Fixed

**Pull request merged**